Typing fixes for the Webserver controller

OzGav opened the pull request:
Tested working locally. The problem that caused the webserver to crash after the last PR has been resolved with an update to the models.

OzGav committed:
more mypy fixes

OzGav committed:
Drafting post upstream changes

OzGav committed:
Fix merge conflicts

OzGav committed:
fix run_handler

marcelveldt commented on music_assistant/controllers/webserver.py:
```diff
@@ -130,7 +128,7 @@ async def get_config_entries(
         """Async initialize of module."""
         # work out all routes
-        routes: list[tuple[str, str, Awaitable]] = []
+        routes: list[tuple[str, str, Any]] = []
```
this should probably be types as a callable returning a coroutine

OzGav replied:
There is getting beyond my level of knowledge but Claude says!
"I used Awaitable[web.StreamResponse] to match aiohttp's Handler type alias. Did you want the more specific Coroutine[Any, Any, web.StreamResponse] instead? Awaitable is the broader type that includes coroutines and other awaitables."

Thoughts?

marcelveldt replied:
I think Awaitable[web.StreamResponse] is fine!

marcelveldt reviewed (approved):
LGTM but needs good testing as last time this broke the entire webserver. Lets release in nightly so we can do a bit testing before it goes to beta

OzGav committed:
PR review comment

OzGav committed:
Merge branch 'dev' into webserver-mypy

OzGav committed:
Merge conflict fixes

OzGav commented:
Since the approval changes made to lines 281-283 and line 334 due to merge conflicts.

OzGav committed:
Fix WebserverController __init__ signature to match CoreController

OzGav commented:
And yes the type fixes did break something! THIS DEVICE wasn't working due to a init signature mismatch with the core controller. This is now fixed. Changed lines 64 and 66

MarvinSchenkel commented:
Just did some quick testing and the webserver is looking fine at first sight. Let's do some more testing in the nightly
